### PR TITLE
[dbnode] Add ability to write existing FST segment out

### PR DESCRIPTION
**Pull request by robskillington**
**What this PR does / why we need it**:

Adds ability to write existing FST segments out to disk by writing out the segment data.

Also adds the ability to write out in memory segments to a temporary file for inspection.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/dbnode/storage/index.go [hunk 1]
```diff
@@ -1588,6 +1588,67 @@ func (i *nsIndex) CleanupDuplicateFileSets() error {
...
+		SetFilePathPrefix(opts.OutputDirectory)
+
+	for _, block := range i.state.blocksByTime {
+		segmentsData, err := block.MemorySegmentsData(ctx)
+		if err != nil {
+			return err
+		}
+
+		for numSegment, segmentData := range segmentsData {
+			indexWriter, err := fs.NewIndexWriter(fsOpts)
```
Comment (notbdu) [inline src/dbnode/storage/index.go:1613]: Do we care to reuse this across segments?

File: src/dbnode/storage/index/block.go [hunk 2]
```diff
@@ -1499,6 +1499,37 @@ func (b *block) EvictMutableSegments() error {
 	return multiErr.FinalError()
 }
 
+func (b *block) MemorySegmentsData(ctx context.Context) ([]fst.SegmentData, error) {
+	b.Lock()
+	defer b.Unlock()
```
Comment (notbdu) [inline src/dbnode/storage/index/block.go:1504]: Could this be a read lock instead?

File: src/dbnode/storage/index.go (hunk 1 above)
Comment (robskillington) [reply src/dbnode/storage/index.go:1613]: They're not reuseable at this time, heh.

File: src/dbnode/storage/index/block.go (hunk 2 above)
Comment (robskillington) [reply src/dbnode/storage/index/block.go:1504]: Yes true, will update.

**Pull request merged**